### PR TITLE
Add PININT API

### PR DESCRIPTION
**Pull request by hannobraun**
This pull request is based on #226. I recommend reviewing/merging that one first.

### CODE REVIEW

**Comment by hannobraun** [comment]
Rebased, ready for review.

File: src/pinint/traits.rs [hunk 1]
```diff
@@ -0,0 +1,16 @@
+/// Implemented by types that identify pin interrupts
+///
+/// This trait is an internal implementation detail and should neither be
+/// implemented nor used outside of LPC8xx HAL. Any changes to this trait won't
+/// be considered breaking changes.
+pub trait Trait {
```
Comment (david-sawatzke) [inline src/pinint/traits.rs]: We should probably look into sealed traits for the future (Should probably not be fixed in this PR, to have a consistent style)

File: src/lib.rs [hunk 2]
```diff
@@ -375,10 +383,17 @@ pub struct Peripherals {
 
     /// Pin interrupt and pattern match engine
     ///
-    /// A HAL API for this peripheral has not been implemented yet. In the
-    /// meantime, this field provides you with the raw register mappings, which
-    /// allow you full, unprotected access to the peripheral.
-    pub PINT: pac::PINT,
+    /// A HAL API for this peripheral has not been implemented yet for LPC8x. In
```
[the remark this answers is not in the document]
Comment (hannobraun) [reply src/lib.rs]: Yes, thanks.

Let me try something:
```suggestion
    /// A HAL API for this peripheral has not been implemented yet for LPC82x. In
```

File: src/pinint/traits.rs (hunk 1 above)
Comment (hannobraun) [reply src/pinint/traits.rs]: Yes, definitely.

I have been aware of that technique for a while, and the reason I haven't done it for the newer traits is just pure laziness (which isn't a valid reason for anything). I'll keep it in mind for future traits, and if I happen to do more cleanup operations, I'll fix older traits to, as I go.

**Comment by hannobraun** [comment]
Review comment applied. Merging.

**Pull request merged**